### PR TITLE
Some steps to more modern CMake

### PR DESCRIPTION
**Pull request by jrood-nrel**
This is a first step to localizing source and include files. This seems to work by merely removing the global source file property, enforcing CMake >= 3.13, and using `target_sources`.

@sayerhs I started to look at using `target_include_directories` in the `src` directory here, but since there is currently one large `include` directory, would we have to merge the `include` directory with the `src` directory to localize the include paths per directory? Or would we just cut off the directory prefixes and use relative paths in the `.C` `#include` lines and use the explicit paths in `target_include_directories` that pair each `src` subdirectory with its `include` subdirectory? Hard to explain my thought with that correctly I think.

I just realized we also don't seem to need a C compiler here so I removed that.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -236,22 +236,6 @@ configure_file("${CMAKE_SOURCE_DIR}/cmake/NaluVersionInfo.h.in"
...
-   endforeach()
-endfunction(add_sources)
-add_subdirectory(src)
-add_subdirectory(unit_tests)
-get_property(NALU_SOURCES GLOBAL PROPERTY GlobalSourceList)
-get_property(NALU_UNIT_SOURCES GLOBAL PROPERTY GlobalUnitSourceList)
-
 include_directories(${CMAKE_SOURCE_DIR}/include)
 include_directories(${CMAKE_BINARY_DIR}/include)
 add_library(nalu ${NALU_SOURCES})
```
Comment (sayerhs) [inline CMakeLists.txt]: This line can just be `add_library(nalu)`? Similarly with `add_executable` for `unittest`

Comment (jrood-nrel) [reply CMakeLists.txt]: Ah, yeah I noticed that after I submitted this.

**Comment by jrood-nrel** [comment]
I made a few more changes to localize to the `nalu` target. We'll talk this over before I merge it though. I have a somewhat loose understanding of exactly what I'm doing with the latest commit but it compiled for me.

**Comment by jrood-nrel** [comment]
Ok, I think I'm going to merge this today, but I will keep a close eye on if this will break compiling for anyone. I will try to broadcast that we will require CMake 3.13 for this.

**Pull request merged**